Demo/VIper-v2-Demo/Viper-v2-Demo/Modules/Home/HomeViewController.swift

```diff
@@ -15,16 +15,83 @@ final class HomeViewController: UIViewController {
     // MARK: - Public properties -
 
     var presenter: HomePresenterInterface!
+    
+    @IBOutlet weak var emptyPlaceholderLabel: UILabel!
+    @IBOutlet weak var tableView: UITableView!
+    @IBOutlet weak var activityIndicator: UIActivityIndicatorView!
 
     // MARK: - Lifecycle -
 
     override func viewDidLoad() {
         super.viewDidLoad()
+        
+        _setupView()
+        presenter.viewDidLoad()
+    }
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        navigationController?.setNavigationBarHidden(false, animated: animated)
+    }
+    
+    private func _setupView() {
+        emptyPlaceholderLabel.isHidden = true
+        tableView.tableFooterView = UIView(frame: CGRect.zero)
+    }
+    
+    @IBAction func logoutButtonActionHandler(_ sender: Any) {
+        presenter.didSelectLogoutAction()
+    }
+    
+    @IBAction func addButtonActionHandler(_ sender: Any) {
+        presenter.didSelectAddAction()
     }
-	
 }
 
 // MARK: - Extensions -
 
 extension HomeViewController: HomeViewInterface {
+    
+    func reloadData() {
+        tableView.reloadData()
+    }
+    
+    func setLoadingVisible(_ visible: Bool) {
+        if visible {
+            activityIndicator.startAnimating()
+        } else {
+            activityIndicator.stopAnimating()
+        }
+    }
+    
+    func setEmptyPlaceholderHidden(_ hidden: Bool) {
+        emptyPlaceholderLabel.isHidden = hidden
+    }
+    
+}
+
+extension HomeViewController: UITableViewDataSource {
+    
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return presenter.numberOfSections()
+    }
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return presenter.numberOrItems(in: section)
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: "cell", for: indexPath) as! HomeTableViewCell
```

Comment:
Please consider better identifier naming, like `HomeTableViewCell` since it is common practice that cell identifier is same as class name.

Reply:
You can use this simple extension to make things prettier.
```
extension UITableView {
   func dequeueReusableCell<T: UITableViewCell>(ofType type: T.Type, withReuseIdentifier identifier: String? = nil, for indexPath: IndexPath) -> T {
        let identifier = identifier ?? String(describing: type)
        return dequeueReusableCell(withIdentifier: identifier, for: indexPath) as! T
    }
}
```